### PR TITLE
Add same_as to NodeBase

### PR DESCRIPTION
**Pull request by wweic**
1. Most class inherited from NodeBase(Schedule, Stage, etc) still have
the convenience of using '==' for object identity. And this is the right
behavior for non-Expr classes.
2. subclasses of ExprOp now create EQ expression when '==' is used.

`__nonzero__` and `__bool__` in EQ and NE is a comprise that in some cases
object identity semantics is still useful, like in unit test. For instance:
````
assert a == b
````

"a == b" will create EQ expression, assert then calls `__nonzero__` of the
result expression. `Expr.__nonzero__` throws exception since it prohibits
evaluating IR expression.

More complex case like:
````
assert a in b # b is dict
````

it will call `__eq__` on a and all keys of b, then `__bool__` on the result
expression. This could not easily be done by same_as.

### CODE REVIEW

File: python/tvm/expr.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def astype(self, dtype):
         return _make.static_cast(dtype, self)
 
 
-class Expr(NodeBase, ExprOp):
+class Expr(ExprOp, NodeBase):
```
Comment (wweic) [inline python/tvm/expr.py]: This relies on [MRO](https://www.python.org/download/releases/2.3/mro/) to select `__eq__`  from `ExprOp`.

**Comment by tqchen** [comment]
Seems quite a lot of unit tests did not pass under python3(python2 version all passes)

**Comment by wweic** [comment]
Yeah, it's weird since `./tests/scripts/task_python_unittest.sh` passed on my machine. Will take a look later.

**Comment by wweic** [comment]
Now I get a repro.

**Pull request merged**